src/libsync/syncfilestatustracker.cpp

```diff
@@ -77,6 +77,43 @@ SyncFileStatusTracker::SyncFileStatusTracker(SyncEngine *syncEngine)
             this, SLOT(slotItemCompleted(const SyncFileItem&)));
 }
 
+SyncFileStatus SyncFileStatusTracker::rootStatus()
+{
+    /* Possible values for the status:
+    enum SyncFileStatusTag {
+        StatusNone,
+        StatusSync,
+        StatusWarning,
+        StatusUpToDate,
+        StatusError,
+    };
+    */
+    SyncFileStatus status =  SyncFileStatus::StatusUpToDate;
+
+    if( !_syncEngine ) return SyncFileStatus::StatusNone;
+
+    if( _syncEngine->isSyncRunning() ) {
+        status = SyncFileStatus::StatusSync;
+    } else {
+        // sync is not running. Check dirty list and _syncProblems
+        int errs = 0, warns = 0;
+        for (auto it = _syncProblems.begin(); it != _syncProblems.end(); ++it) {
+            if( it->second == SyncFileStatus::StatusError ) {
+                errs ++;
+                break; // stop if an error found at all.
+            } else {
+                warns ++;
+            }
+        }
+        if( errs ) {
+            status = SyncFileStatus::StatusError;
```

Comment:
We only bubble up errors as a warning for parents folder. Since there can be no error with the root folder itself (it is not synced), it should never show an error. The tray icon error on the other hand applies to all files, not just the root.